contracts/PepemonRewardPool.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./lib/RewardPool.sol";
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract PepemonRewardPool is RewardPool, Ownable {
+    struct Reward {
+        uint256 tokenId;
+        address tokenAddress;
+    }
+    Reward[] private rewardPool;
+
+    // Transfer a token from the sender
+    // store it in the pool
+    // if not already done, approve the Matchmaker to transfer the token
+    function addReward(address tokenAddress, uint256 tokenId) external override onlyOwner {
+        require(
+            ERC1155(tokenAddress).balanceOf(address(msg.sender), tokenId) > 0,
+            "PepemonRewardPool: You don't have enough of this token"
+        );
+
+        ERC1155(tokenAddress).safeTransferFrom(address(msg.sender), address(this), tokenId, 1, "");
+        rewardPool.push(Reward(tokenId, tokenAddress));
```

Comment:
We'll need to approve transfers from this pool to the matchmaker contract.

A list of addresses seems like it would be the most flexible.
Something like `addRewardOperator(matchmakerAddress) onlyOwner`
What do you think?


We'd probably need to loop through them when we see a new tokenAddress, and then also loop through approved tokenAddresses when we `removeRewardOperator(matchmakerAddress)`...